src/ripple/app/misc/impl/TxQ.cpp

```diff
@@ -66,31 +114,34 @@ getLastLedgerSequence(STTx const& tx)
 }
 
 static
-std::uint64_t
-increase(std::uint64_t level,
+FeeLevel64
+increase(FeeLevel64 level,
     std::uint32_t increasePercent)
 {
     return mulDiv(
-        level, 100 + increasePercent, 100).second;
+        level, 100u + increasePercent, 100u).second;
 }
 
 
 //////////////////////////////////////////////////////////////////////////
 
+constexpr FeeLevel64 TxQ::baseLevel;
+
 std::size_t
 TxQ::FeeMetrics::update(Application& app,
     ReadView const& view, bool timeLeap,
     TxQ::Setup const& setup)
 {
-    std::vector<uint64_t> feeLevels;
+    std::vector<FeeLevel64> feeLevels;
     auto const txBegin = view.txs.begin();
     auto const txEnd = view.txs.end();
     auto const size = std::distance(txBegin, txEnd);
     feeLevels.reserve(size);
     std::for_each(txBegin, txEnd,
         [&](auto const& tx)
         {
-            auto const baseFee = calculateBaseFee(view, *tx.first);
+            auto const baseFee = view.fees().toDrops(
```

Comment:
I'll need to document this, but the basic idea is that `toDrops` returns a result like the one returned from `mulDiv`. That function is documented. The first part of the pair is a bool indicating whether the result is "safe" (ie. Doesn't overflow). The second is the result of the computation, or the max value if it did overflow. The bool is ignored in many places, because those places just need a value, but it is also checked in several others, at least in the general case of `mulDiv`.